src/prefect/engine/result/base.py

```diff
@@ -207,9 +214,11 @@ def format(self, **kwargs: Any) -> "Result":
             - Result: a new result instance with the appropriately formatted location
         """
         new = self.copy()
-        if new.location is not None:
+        if isinstance(new.location, str):
             assert new.location is not None
```

Comment:
I think this assertion is redundant now that you're checking for the type

Reply:
Yea that assert is purely for `mypy` type checking